### ISSUE TITLE
Server: Make code block tool results visible when `rumdl` is in server mode

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#463]
### Problem or motivation

Code block tools (https://github.com/rvben/rumdl/blob/957ec0a374bc34a87c49653a58b821588233c4e0/docs/code-block-tools.md) extend `rumdl` checks into fenced code blocks.

However, the results of these checks are invisible within an editor that is configured to use `rumdl` in server mode.

For example, this configuration:

```toml
[code-block-tools]
enabled = true

[code-block-tools.languages.markdown]
lint = ["rumdl"]
```

And for example, this Markdown:

````markdown
---
title: Level 1 heading
---

Here is some invalid Markdown:

- List item 1
* List item 2

Here is a fenced code block with invalid Markdown:

```markdown
Here is some invalid Markdown:

- List item 1
* List item 2
```
````

At the moment, within an editor, you can only see a problem at line 8. You must run `rumdl check` at the command line to also see the problem at line 16.

### Proposed solution

Ideally, `rumdl` would show in server mode the same results it shows when run as `rumdl check`.

Within an editor, for configured code block tool languages, fenced code blocks could be flagged as having violations. This would look the same as highlighting of Markdown violations outside of fenced code blocks. Ideally, the specific line with the violation within the fenced code block would be flagged.

### Feature category

LSP / editor integration

**Comment by rvben** [comment]
Code block tool diagnostics are now surfaced in server mode since v0.1.29.

Thanks @eread for the report!
